tests/e2e/utils/service_utils.go

```diff
@@ -0,0 +1,82 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const (
+	ServiceAnnotationDNSLabelName = "service.beta.kubernetes.io/azure-dns-label-name"
+)
+
+// DefaultService returns a default service representation
+func DefaultService(name string) (result *v1.Service, labels map[string]string) {
+	result = &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: labels,
+		},
+	}
+	return
+}
+
+// CreateLoadBalancerService creates a new service with service port
+func CreateLoadBalancerService(c clientset.Interface, name string, labels map[string]string, namespace string, ports []v1.ServicePort) (*v1.Service, error) {
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Annotations: map[string]string{
+				ServiceAnnotationDNSLabelName: name + string(uuid.NewUUID()),
```

Comment:
the generation looks good, but suppose the whole dnslabelname should be passed as parameter.

The test acts as user, he specifies the name and then call with the name, there would be no additional effort in getServiceDNS to retrieve the label